### PR TITLE
Drop vendor registry details from container image concept

### PR DESCRIPTION
**Pull request by sftim**
Minimal rewrite to avoid referencing specific 3rd party vendors and their implementations.

Helps with #20232 and #21641

The amended page *still needs work*; that's OK, that's something we can pick up in a separate pull request.

### CODE REVIEW

**Comment by celestehorgan** [comment]
@sftim to be clear: when you say "The amended page _still needs work_;" – do you want review on grammar/language here or just for me to catch the typos and let you go on your merry way? :)

**Comment by sftim** [comment]
So long as _these changes_ have no regressions, let's merge.

File: content/en/docs/concepts/containers/images.md [hunk 1]
```diff
@@ -9,18 +9,51 @@ weight: 10
 
 {{% capture overview %}}
```
[the remark this answers is not in the document]
Comment (zacharysarah) [reply content/en/docs/concepts/containers/images.md:10]: Alas! Merge conflict, likely from #21359!

**Comment by jimangel** [comment]
Re-triggered the build it had a strange memory error.
/lgtm

**Comment by jimangel** [comment]
The logs:
```
9:29:19 AM: hugo --enableGitInfo --buildFuture -b https://deploy-preview-21438--kubernetes-io-master-staging.netlify.app
9:29:19 AM: Building sites …
9:29:21 AM: panic: runtime error: invalid memory address or nil pointer dereference
9:29:21 AM: [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x109ad12]
9:29:21 AM: goroutine 84 [running]:
9:29:21 AM: github.com/gohugoio/hugo/hugolib.(*shortcodeHandler).extractShortcode(0xc0069b6c80, 0xb, 0x0, 0xc005a11090, 0x0, 0x0, 0x1a14380)
9:29:21 AM: 	/root/project/hugo/hugolib/shortcode.go:505 +0x882
```

```
9:29:21 AM: ┌─────────────────────────────┐
9:29:21 AM: │   "build.command" failed    │
9:29:21 AM: └─────────────────────────────┘
9:29:21 AM: ​
9:29:21 AM:   Error message
9:29:21 AM:   Command failed with exit code 2: make deploy-preview
9:29:21 AM: ​
9:29:21 AM:   Error location
9:29:21 AM:   In build.command from netlify.toml:
9:29:21 AM:   make deploy-preview
9:29:21 AM: ​
9:29:21 AM:   Resolved config
9:29:21 AM:   build:
9:29:21 AM:     command: make deploy-preview
9:29:21 AM:     functions: /opt/build/repo/functions
9:29:21 AM:     publish: /opt/build/repo/public
```

File: content/en/docs/concepts/containers/images.md [hunk 2]
```diff
@@ -375,4 +298,7 @@ common use cases and suggested solutions.
 If you need access to multiple registries, you can create one secret for each registry.
 Kubelet will merge any `imagePullSecrets` into a single virtual `.docker/config.json`
 
+## {{% heading "whatsnext" %}}
 
+* Read the [OCI Image Manifest Specification](https://github.com/opencontainers/image-spec/blob/master/manifest.md)
+{{% /capture %}}
```
Comment (jimangel) [inline content/en/docs/concepts/containers/images.md:304]: I think this is causing the build failure.

Comment (sftim) [reply content/en/docs/concepts/containers/images.md]: Yep! Oops! Fixed!

**Comment by tengqm** [comment]
(re)approve
/lgtm
/approve

**Comment by et304383** [comment]
@sftim What was the rationale for this change and where are the replacement docs that explain that K8s will automatically authenticate against AWS ECR periodically if running on EC2 with the appropriate IAM role permissions?

I just wasted a day trying to understand how K8s was authenticating in AWS when trying to replicate our cluster in OCI, and without this documentation there is NOTHING on the Internet that explains that K8s does this automatically.

**Comment by sftim** [comment]
This change helped with two issues: https://github.com/kubernetes/website/issues/20232 and https://github.com/kubernetes/website/issues/21641. 
Kubernetes itself does **not** periodically authenticate against Amazon ECR. However, EC2 instances running with a particular AMI on AWS may well do that. For details on how a particular cloud provider has set up their flavor of Kubernetes, it is best to check the documentation provided by that cloud provider.

The blog article https://kubernetes.io/blog/2020/05/third-party-dual-sourced-content/ provides more content about why Kubernetes avoids documenting external / third party software.

**Comment by et304383** [comment]
> This change helped with two issues: #20232 and #21641. Kubernetes itself does **not** periodically authenticate against Amazon ECR. However, EC2 instances running with a particular AMI on AWS may well do that. For details on how a particular cloud provider has set up their flavor of Kubernetes, it is best to check the documentation provided by that cloud provider.
Actually, it does, because we're using a non managed Kubernetes on AWS EC2 with IAM roles.

The K8s documentation used to state this.  We're running on version 1.18.  Are you saying this functionality is removed in later versions?

**Comment by sftim** [comment]
The Kubernetes documentation no longer includes documentation for third-party systems and software, other than those components necessary for Kubernetes to run.

It's possible that the kubelet does include an integration with AWS IAM and AWS ECR; if so, that's news to me. We'd be happy to document that functionality (perhaps in a different way) if it does exist _as part of Kubernetes itself_ because that would then be built in to Kubernetes.

**Pull request merged**